tests/queries/0_stateless/00493_substring_of_enum.sql

```diff
@@ -0,0 +1,32 @@
+DROP TABLE IF EXISTS tab;
+CREATE TABLE tab(e8 Enum8('hello' = -5, 'world' = 15), e16 Enum16('shark' = -999, 'eagle' = 9999)) ENGINE MergeTree ORDER BY tuple();
+INSERT INTO TABLE tab VALUES ('hello', 'shark'), ('world', 'eagle');
+
```

Comment:
Such tests are smart but 1. they are difficult to debug in case something goes wrong and 2. they test unrelated functionality (joins) which can potentially break the test if something is wrong with joins.

My personal preference are much more "dumber tests", basically:
- check that the preconditions are checked (see `getReturnTypeImpl()`, e.g. one could test that `substringUTF8` with enum argument throws)
- then make a testing matrix of
1. const vs. non-const inputs (use materialize() or CREATE TABLE)
2. offset < 0 vs. offset == 0 vs. offset > 0 3. length < 0, length == 0, length > 0

But this is not a blocker, if you like we can keep the existing test.

Reply:
I will make them dumber :) Thanks.

Reply:
https://github.com/ClickHouse/ClickHouse/pull/57821